File: base/src/main/java/com/tsy/base/api/error/DefaultErrorBundle.java
package com.tsy.base.api.error;

/**
 * Created by jay on 17/8/1.
 */

public class DefaultErrorBundle implements ErrorBundle {

    private static final String DEFAULT_ERROR_MSG = "Unknow error";

    private final Exception exception;

    public DefaultErrorBundle(Exception exception) {
        this.exception = exception;
    }

    @Override
    public Exception getException() {
        return exception;
    }

    @Override
    public String getErrorMessage() {
        return (exception != null) ? this.exception.getMessage() : DEFAULT_ERROR_MSG;
    }
}

File: base/src/main/java/com/tsy/base/api/error/DefaultErrorBundle.kt
package com.tsy.base.api.error

/**
 * Created by jay on 17/8/1.
 */

class DefaultErrorBundle(private val exception: Exception?) : ErrorBundle {

    override fun getException(): Exception? {
        return exception
    }

    override fun getErrorMessage(): String {
        return if (exception != null) this.exception.message!! else DEFAULT_ERROR_MSG
    }

    companion object {

        private val DEFAULT_ERROR_MSG = "Unknow error"
    }
}
